apps/ledger-live-mobile/src/components/Nft/NftList.tsx

```diff
@@ -67,30 +86,123 @@ export function NftList({ data }: Props) {
+    ],
   );
 
+  const gradients = colors.type === "light" ? lightGradients : darkGradients;
   return (
     <>
       <TrackScreen category="NFT Gallery" NFTs_owned={data.length} />
 
       <RefreshableCollapsibleHeaderFlatList
         numColumns={2}
+        ListHeaderComponent={
+          <Animated.View>
+            {!onMultiSelectMode && (
+              <Animated.View entering={FadeInDown} exiting={FadeOutDown}>
+                <Flex
+                  width="100%"
+                  flexDirection="row"
+                  alignItems="center"
+                  justifyContent="flex-start"
+                >
+                  <StyledButton
+                    onPress={multiSelectModeAction}
+                    type="default"
+                    iconName="Tasks"
+                    iconPosition="left"
+                    size="small"
+                  >
+                    <Text variant="body" fontWeight="semiBold">
+                      {t("wallet.nftGallery.filters.selectAndHide")}
+                    </Text>
+                  </StyledButton>
+                </Flex>
+              </Animated.View>
+            )}
+          </Animated.View>
+        }
+        ListHeaderComponentStyle={{
+          marginBottom: onMultiSelectMode ? 0 : space[6],
+        }}
         data={dataWithAdd}
         renderItem={renderItem}
         keyExtractor={keyExtractor}
         showsVerticalScrollIndicator={false}
         initialNumToRender={6}
         windowSize={11}
-        contentContainerStyle={{ marginTop: 16 }}
+        contentContainerStyle={{ marginTop: 0, marginHorizontal: space[6] }}
         testID={"wallet-nft-gallery-list"}
       />
+      <Animated.View>
+        {onMultiSelectMode && (
+          <Animated.View entering={FadeInDown} exiting={FadeOutDown}>
+            <BackgroundGradient {...gradients[0]} />
+
+            <ButtonsContainer width="100%" justifyContent={"space-between"}>
+              <StyledButton
+                onPress={onClickHide}
+                type="main"
+                iconName="EyeNone"
+                iconPosition="left"
+                size="large"
+                flexGrow={1}
+                disabled={nftsToHide.length === 0}
+              >
+                {t("wallet.nftGallery.filters.hide", {
+                  count: nftsToHide.length,
+                })}
+              </StyledButton>
+              <StyledButton
+                onPress={readOnlyModeAction}
+                type="default"
+                iconPosition="left"
+                size="large"
+                flexGrow={1}
+              >
+                {t("common.cancel")}
+              </StyledButton>
+            </ButtonsContainer>
+          </Animated.View>
+        )}
+      </Animated.View>
     </>
   );
 }
+
+const StyledButton = styled(Button)`
+  padding: 0;
+  margin: 0;
+`;
+
+const ButtonsContainer = styled(Flex)`
+  position: absolute;
+  bottom: 20px;
+  z-index: 5;
+  padding: 0 18px;
```

Comment:
please no hardcoded px value, use a value from the design system (take the closest one if it doesn't exist and give feedback to designers)